tests/grid_test.py

```diff
@@ -1,126 +1,357 @@
 import numpy
 import pytest
+import scipy.stats.qmc
 
-from smolyay.basis import (ChebyshevFirstKind, BasisFunctionSet, 
-                           NestedBasisFunctionSet)
-from smolyay.grid import (IndexGridGenerator, SmolyakGridGenerator,
-                          TensorGridGenerator, generate_compositions)
+import smolyay
 
 
-def test_error_basis_set_type():
-    """Test if raise TypeError works for IndexGridGenerator."""
-    basis_set = "not a BasisFunctionSet"
+@pytest.mark.parametrize(
+    "random_point_set",
+    [
+        smolyay.samples.UniformRandomPointSet,
+        smolyay.samples.LatinHypercubeRandomPointSet,
+        smolyay.samples.HaltonRandomPointSet,
+        smolyay.samples.SobolRandomPointSet,
+    ],
+    ids=["Uniform", "Latin", "Halton", "Sobol"],
+)
+def test_random_initalize(random_point_set):
+    """Test that the random point set initializes correctly"""
+    f = random_point_set([[-10, 10], [0, 2]], 64, 1234)
+    assert numpy.array_equal(f.domain, [[-10, 10], [0, 2]])
+    assert f.num_dimensions == 2
+    assert f.num_points == 64
+    assert isinstance(f.num_points, int)
+    assert f.seed == 1234
+    assert isinstance(f.seed, int)
+    f.domain = [-10, 10]
```

Comment:
Make sure you use white space to make things readable. I almost missed this was testing both 2d and 1d domains & the setters
```suggestion

    f.domain = [-10, 10]
```